src/jvm/test/org/jetbrains/kotlinx/lincheck_test/transformation/channel/IncorrectBufferedChannel.kt

```diff
@@ -0,0 +1,772 @@
+/*
+ * Lincheck
+ *
+ * Copyright (C) 2019 - 2024 JetBrains s.r.o.
+ *
+ * This Source Code Form is subject to the terms of the
+ * Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed
+ * with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.jetbrains.kotlinx.lincheck_test.transformation.channel
+
+import kotlinx.atomicfu.*
+import kotlinx.coroutines.CancellableContinuation
+import kotlinx.coroutines.InternalCoroutinesApi
+import kotlinx.coroutines.suspendCancellableCoroutine
+
+/**
+ * Incorrect implementation of the BufferedChannel, intentionally provided by Daria Shutina.
```

Comment:
Please add a reference to the issue, and put the code and the test in the same file.